pkg/cmd/gist/clone/clone_test.go

```diff
@@ -60,32 +63,32 @@ func Test_GistClone(t *testing.T) {
 		{
 			name: "shorthand",
 			args: "GIST",
-			want: "git clone https://gist.github.com/GIST.git",
+			want: `git -c credential.helper= -c credential.helper=!"some/path/gh" auth git-credential clone https://gist.github.com/GIST.git`,
```

Comment:
It's potentially worrying that this implementation detail is now added to the expectation in so many tests unrelated to authentication itself. If we ever change a single character of this, all these tests will also have to be updated.

Should these test assertions attempt to normalize the git invocation line with a single shared helper that can be kept to sync with the authentication implementation?

Reply:
Yeah I agree it is not great. It is a product of how these tests were written in the past. Two solutions I can think of:
1. Modify all these commands to use an interface which is fulfilled by `git.Client` and then in tests mock out the method calls. This would be a good amount of work, but is also probably the most "go" way.
2. Add a function to the `CommandStubber` specific for registering `git` commands which would create a single place where the authentication string would be. 
What are your thoughts?

Reply:
Agreed option (1) would be the most "correct" but perhaps infeasible for this PR. How about option 3: make a test helper that converts a string like `git -c credential.helper= -c credential.helper=... <command> <args>` to `git <command> <args>` and use that in equality assertions?

Reply:
I don't necessarily like that approach as it weakens the assertions about the git commands being invoked and would no longer assert that `gh` was set as the credential helper, which is the purpose of this PR.